vm_core.h

```diff
@@ -362,6 +362,7 @@ struct rb_iseq_constant_body {
 	    unsigned int ambiguous_param0 : 1; /* {|a|} */
 	    unsigned int accepts_no_kwarg : 1;
             unsigned int ruby2_keywords: 1;
+            unsigned int defined_in_eval: 1;
```

Comment:
This is not parameter information.

Reply:
True.  If that's a problem, the flag could be moved.  It just doesn't require additional storage space if placed there.  Maybe saving a few bytes isn't worth it?

Reply:
I'm planning to introduce this information in `location`.

Reply:
That sounds good.  I'll leave this pull request open, hopefully it will be helpful and save you some effort.

Reply:
Thank you. Now I tried to put information in `location::realpath` (make it nil for eval context). I think `realpath == nil` is reasonable because it is not checked. 

However `__dir__` and `require_relative` uses it. There is no space in a `T_STRING`, so I'm thinking again.

Options:

(1) Make realpath nil and `__dir__` and `require_relative` uses path instead. However, there is incompatible:

```ruby
binding.eval('p __FILE__; p __dir__')
#=> "(eval)"
#=> nil
binding.eval('p __FILE__; p __dir__', '(eval)')
#=> "(eval)"
#=> "."

# proposed:
binding.eval('p __FILE__; p __dir__')
#=> "(eval)"
#=> "."
```

(2) Pass Qtrue or some value instead of String as a realpath to specify the path is passed by eval.
`__dir__` can check `path`. It seems reasonable, but need to re-check all realpath reference code.

(3) Pass realpath string with a flag. No idea now how to add a flag.
(3) others.

Reply:
I want to believe that nobody relies on `ISeq#to_a` result.